### PR TITLE
Documentation and features for setup as conda package

### PR DESCRIPTION
**Pull request by CalCraven**
This PR builds out the API for this package. Additionally, doc strings
were added for ease of install and operations. Changes are as follows:

* Updated README.md docs for installation and quick use
 instructions.
 * Rename core module from prepackaged.py to xml_loader.py.
 * Refactor xml_loader.FoyerFFs class to allow for single use loading
of xml information, or loading and storing xml.
 * Added xml_loader.load_xml function to get xml info from local
 *  path or from the xml directory within foyer.
 * Added Error messages during loading steps.
 * Updated unit testing to use updated api.

### CODE REVIEW

File: forcefield_utilities/tests/test_gmso_ff.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from gmso.utils._constants import FF_TOKENS_SEPARATOR
 
 from forcefield_utilities.foyer_xml import AtomTypes, NonBondedForce
-from forcefield_utilities.prepackaged import FoyerFFs
+from forcefield_utilities  import FoyerFFs
```
Comment (umesh-timalsina) [inline forcefield_utilities/tests/test_gmso_ff.py:7]:
```suggestion
from forcefield_utilities.xml_loader  import FoyerFFs
```

File: forcefield_utilities/xml_loader.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            self.loaded_ffs[ffname] = self.ff_registry[ffname]()
+        else:
+            self.loaded_ffs[ffname] = self.ff_registry["custom_xml"](
+                ffname, rel_to_module
+            )
+        return self.loaded_ffs[ffname]
+
+    def __getitem__(self, ffname):
+        """Get function for indexing by loaded forcefields"""
+        if not ffname in self.loaded_ffs.keys():
```
Comment (umesh-timalsina) [inline forcefield_utilities/xml_loader.py:156]:
```suggestion
        if ffname not in self.loaded_ffs.keys():
```

File: forcefield_utilities/foyer_xml.py [hunk 3]
```diff
@@ -621,7 +622,10 @@ def iterate_on(self, children_type):
                 if type(child) == loaders[children_type]:
                     for type_children in child.children:
                         yield type_children
-
+    """TODO:
+    Need a to_foyer_ff method.
+    Will be similar the to_gmso_ff here.
+    """
```
Comment (umesh-timalsina) [inline forcefield_utilities/foyer_xml.py:628]: This can be incorporated separately.
```suggestion
```

File: forcefield_utilities/xml_loader.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        "trappe_ua": load_trappe_ua,
+        "custom_xml": load_xml,
+    }
+
+    def __init__(self):
+        super().__init__()
+        self.loaded_ffs = {}
+
+    @classmethod
+    def GetFF(self, ffname, rel_to_module=False):
```
Comment (umesh-timalsina) [inline forcefield_utilities/xml_loader.py:108]:
```suggestion
    def get_ff(self, ffname, rel_to_module=False):
```

**Review by umesh-timalsina — CHANGES_REQUESTED** [review]

File: forcefield_utilities/foyer_xml.py (hunk 3 above)
Comment (CalCraven) [reply forcefield_utilities/foyer_xml.py]: Addressed in commit 1dcd2fc

**Comment by CalCraven** [comment]
pre-commit.ci autofix

**Comment by CalCraven** [comment]
"pre-commit.ci autofix"

**Comment by CalCraven** [comment]
pre-commit.ci run

**Review by umesh-timalsina — APPROVED** [review]

**Pull request merged**